onsite/ascore.py

```diff
@@ -1,283 +1,202 @@
     
     def updateMembers_(self):
-        """Update member variables from the parameter object."""
-        self.fragment_mass_tolerance_ = self.getParameter("fragment_mass_tolerance").toDouble()
-        self.fragment_tolerance_ppm_ = (self.getParameter("fragment_mass_unit").toString() == "ppm")
-        self.max_peptide_length_ = self.getParameter("max_peptide_length").toInt()
-        self.max_permutations_ = self.getParameter("max_num_perm").toInt()
-        self.add_decoys_ = self.getParameter("add_decoy").toBool()
+        """Update member variables from parameter object."""
+        pass
+    
+    def setAddDecoys(self, add_decoys):
+        """Set whether to include A (PhosphoDecoy) sites."""
+        self.add_decoys_ = add_decoys
     
     @staticmethod
     def isPhosphoDecoySite(residue):
-        """Check if a residue is a PhosphoDecoy site (A)."""
+        """Check if residue is a PhosphoDecoy site (A)."""
         return residue == 'A'
     
     @staticmethod
     def isPhosphoSite(residue):
-        """Check if a residue is a standard phosphorylation site (S, T, Y)."""
+        """Check if residue is a phosphorylation site (S, T, Y)."""
         return residue in ['S', 'T', 'Y']
     
     def compute(self, hit, real_spectrum):
         """
-        Compute the AScore and return a modified PeptideHit with phosphorylation site scores.
-        
-        Parameters:
-        -----------
-        hit : PeptideHit
-            The peptide hit containing the sequence with potential phosphorylation sites
-        real_spectrum : MSSpectrum
-            The experimental MS/MS spectrum
-            
-        Returns:
-        --------
-        PeptideHit
-            A modified PeptideHit with AScore values for each phosphorylation site
-            The score field contains the best AScore value
+        Compute AScore and return modified PeptideHit with phosphorylation site scores.
         """
         phospho = PeptideHit(hit)
         
-        # reset phospho
-        phospho.setScore(-1)
+        phospho.setScore(-1.0)
         phospho.setMetaValue("search_engine_sequence", hit.getSequence().toString())
         
-        # Early termination for empty spectra
         if real_spectrum.size() == 0:
-            print("Warning: Empty spectrum provided to AScore::compute. Returning original hit.")
             return phospho
         
         sequence_str = phospho.getSequence().toString()
         unmodified_sequence_str = phospho.getSequence().toUnmodifiedString()
         
-        # Count phosphorylation events and remove phosphorylations to get the base peptide sequence
         number_of_phosphorylation_events = self.numberOfPhosphoEvents_(sequence_str)
         seq_without_phospho = self.removePhosphositesFromSequence_(sequence_str)
         
-        # Initialize counters for regular and decoy phosphorylation events
         regular_phospho_count = sequence_str.count("(Phospho)")
```

Comment:
You can access the modification in the AaSequence data structure. No need for wasteful generation of a string and then preparing everything.